src/fixate/drivers/dmm/keithley_6500.py

```diff
@@ -0,0 +1,340 @@
+            )
+
+        self._write(f":COUN {val}")
+        self._is_error()
+        self._samples = val
+
+    def local(self):
+        # Keithley does not have a way to release the DMM from remote mode
+        # So just set up a trigger loop. Requires *TRG to be sent to drop it out of the loop (call to remote()).
+        self.samples = 1
+        self._write("TRIG:LOAD 'EMPTY'")  # Load empty model
+        self._write(f"TRIG:BLOC:MDIG 1, 'defbuffer1', 1")
+        self._write(f"TRIG:BLOC:DEL:CONS 2, 0.1")
+        self._write("TRIG:BLOC:BRAN:EVEN 3, COMM, 5")
+        self._write(f"TRIG:BLOC:BRAN:ALW 4, 1")
+        self._write("TRIG:BLOC:BUFF:CLE 5")
+        self._write("TRAC:CLE")
+        self._write("INIT")
+
+    def remote(self):
+        # Stop trigger loop and return to normal
+        self._write("*TRG")
+        self._is_error()
+
+    def set_manual_trigger(self, samples=1):
+        """
+        Setup instrument for manual triggering
+        :param samples: Number of samples to take per trigger
+
+        Use trigger_measurement() to trigger a measurement.
+        Use measurements() to retrieve measurements.
+        """
+        self._manual_trigger = True
+        self.samples = samples
+        self._write("TRIG:LOAD 'EMPTY'")  # Load empty model
+        self._write(f"TRIG:BLOC:MDIG 1, 'defbuffer1', {samples}")
+        self._write("TRAC:CLE")
+        self._is_error()
+
+    def trigger(self):
+        """
+        Manually trigger a measurement and store in instrument buffer.
+        """
+        self._write("INIT; *WAI")
+        self._is_error()
+
+    def measurement(self):
+        """
+        Sets up DMM triggering, creates list of measurements from the read buffer
+        returns: a single value as a float
+        """
+        return self.measurements()[0]
+
+    def measurements(self):
+        if not self.mode:
+            raise InstrumentError("Please set DMM mode before taking a measurement!")
+
+        with self.lock:
+            return self._read_measurements()
+
+    def reset(self):
+        """
+        Checks for errors and then returns DMM to power up state
+        """
+        self._is_error(silent=True)
+        # Wait for previous commands to finish, reset, clear event logs
+        self._write("*RST")
+        self._CLEAN_UP_FLAG = False
+        self._is_error()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.instrument.close()
+        self.is_connected = False
+
+    def _write(self, data):
+        """
+        Writes data to the DMM
+        raise: ParameterError if called with no data string
+        """
+        if data:
+            if isinstance(data, str):
+                self.instrument.write(data)
+                time.sleep(0.05)
+            else:
+                for itm in data:
+                    self.instrument.write(itm)
```

Comment:
There's a sleep after the write above, but not this one? We only sleep at the end of the loop. Is this intentional? Makes me wonder even more if the previous sleep is even needed.
Also, we've landed in where because `data` wasn't a string, but what if it's an in or list of ints, or anything else? I imagine we're wanting it to to be a list of strings, what gets raised if it gets something that's not?
Perhaps the below would be more robust?
```
if not isinstance(data, str) and not all([isinstance(itm, str) for itm in data]):
    raise ParameterError("...")
```

Reply:
That would need a bit more thought put into it, because it will break in some cases. But I think that general idea would be more robust.

Reply:
Do we have issues with robustness at the moment for this? The only way this function breaks is if something calling it internally sends bad data, seeing as the _write function is not meant to be used externally. 

If you really want this changed, then that's fine, I just think we're fighting small battles here.

Reply:
Now raises a ParameterError if data is not str or list of str